unit/java_bytecode/java_string_library_preprocess/convert_exprt_to_string_exprt.cpp

```diff
@@ -38,22 +38,35 @@ TEST_CASE("Convert exprt to string exprt")
   preprocess.add_string_type("java.lang.String", symbol_table);
   symbol_typet java_string_type("java::java.lang.String");
   symbol_exprt expr("a", pointer_type(java_string_type));
-  convert_exprt_to_string_exprt_unit_test(
+  refined_string_exprt string_expr=convert_exprt_to_string_exprt_unit_test(
     preprocess, expr, loc, symbol_table, code);
+
+  REQUIRE(string_expr.id()==ID_struct);
+  REQUIRE(is_refined_string_type(string_expr.type()));
+
   register_language(new_java_bytecode_language);
 
   std::vector<std::string> code_string;
+  const std::regex spaces("\\s+");
+  const std::regex numbers("\\$[0-9]*");
   for(auto op : code.operands())
-    code_string.push_back(from_expr(ns, "", op));
-
-  REQUIRE(code_string.size()==7);
-  REQUIRE(code_string[0]=="int cprover_string_length$1;");
-  REQUIRE(code_string[1]=="char cprover_string_array$2[INFINITY()];");
-  REQUIRE(code_string[2]=="cprover_string_length$1 = a->length;");
-  REQUIRE(code_string[3]=="__CPROVER_assume(!(a->data == null));");
-  REQUIRE(code_string[4]=="cprover_string_array$2 = *a->data;");
-  REQUIRE(code_string[5]=="struct __CPROVER_refined_string_type { int length; "
-    "char content[INFINITY()]; } cprover_string$3;");
-  REQUIRE(code_string[6]=="cprover_string$3 = { .=cprover_string_length$1, "
-    ".=cprover_string_array$2 };");
+  {
+    const std::string line=from_expr(ns, "", op);
+    code_string.push_back(
+      std::regex_replace(std::regex_replace(line, spaces, " "), numbers, ""));
+  }
+
+  const std::vector<std::string> reference_code=
+    {
+      "char *cprover_string_content;",
+      "int cprover_string_length;",
+      "cprover_string_length = a->length;",
+      "cprover_string_content = a->data;"
+    };
+
+
+  for(std::size_t i=0; i<code_string.size() && i<reference_code.size(); ++i)
```

Comment:
I think this can be simplified to:
```C++
REQUIRE(
  std::equal(std::begin(code_string), std::end(code_string), std::begin(reference_code)));
```
Though I suppose you'd lose some info on which ones are wrong so maybe not a good idea.